sycl/doc/extensions/SPIRV/SPV_INTEL_float_controls2.asciidoc

```diff
@@ -0,0 +1,140 @@
+
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2020-04-24
+| Revision           | 2
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.5, Revision 2, Unified
+
+This extension requires SPIR-V 1.0.
+
+Overview
+--------
+
+This extension provides new execution modes to control floating-point computations by overriding an implementation’s default behavior for rounding modes and floating point modes.
+
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_float_controls2"
+----
+
+Modifications to the SPIR-V Specification, Version 1.5, Revision 2, Unified
+---------------------------------------------------------------------------
+
+Modify Section 3.6, Execution Mode, add the following rows to the Execution Mode table:
+
+--
+[cols="1,20,10,10",options="header",width = "80%"]
+|====
+  2+^| Execution Mode  | Enabling Capabilities | Extra Operands
+| 5620 | *RoundingModeRTPINTEL* +
+The default rounding mode for floating-point arithmetic and conversions instructions must be round to positive infinity.
+If an instruction is decorated with *FPRoundingMode* or defines a rounding mode in its description, that rounding mode is applied and *RoundingModeRTPINTEL* is ignored.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| *RoundToInfinityINTEL* | <<Literal, 'Literal'>>  _Target Width_
+| 5621 | *RoundingModeRTNINTEL* +
+The default rounding mode for floating-point arithmetic and conversions instructions must be round to negative infinity.
+If an instruction is decorated with *FPRoundingMode* or defines a rounding mode in its description, that rounding mode is applied and *RoundingModeRTPNINTEL* is ignored.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| *RoundToInfinityINTEL* | <<Literal, 'Literal'>> _Target Width_
+| 5622 | *FloatingPointModeALTINTEL* +
+The default floating-point operation mode for floating-point arithmetic and conversions instructions must be the Alternative Floating-Point Mode as specified by the client API.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| *FloatingPointModeINTEL* | <<Literal, 'Literal'>> _Target Width_
+| 5623 | *FloatingPointModeIEEEINTEL* +
+The default floating-point operation mode for floating-point arithmetic and conversions instructions must be the IEEE Mode as specified by the client API.
```

Comment:
It is not clear what "the IEEE Mode" is. Probably the IEEE 754 standard is meant here. Maybe it is worth adding reference to the standard as it is done in the OpenCL  SPIR-V environment spec - https://www.khronos.org/registry/OpenCL/specs/2.2/html/OpenCL_Env.html#ieee-754-spec

Reply:
Yes, I will change to IEEE 754

Reply:
Changed to IEEE 754